eureka2-ext/eureka2-eureka1-rest-api/src/main/java/com/netflix/eureka2/eureka1/rest/query/Eureka2FullFetchWithDeltaView.java

```diff
@@ -0,0 +1,162 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.functions.Func1;
+import rx.subjects.BehaviorSubject;
+
+import static com.netflix.eureka2.eureka1.rest.model.Eureka1ModelConverters.toEureka1xApplications;
+import static com.netflix.eureka2.eureka1.rest.model.Eureka1ModelConverters.toEureka1xInstanceInfo;
+import static com.netflix.eureka2.eureka1.rest.model.Eureka1ModelConverters.toEureka1xInstanceInfos;
+import static com.netflix.eureka2.eureka1.rest.model.Eureka1ModelConverters.v1InstanceIdentityComparator;
+import static com.netflix.eureka2.interests.ChangeNotifications.emitAndAggregateChanges;
+import static com.netflix.eureka2.interests.ChangeNotifications.collapseAndExtract;
+import static com.netflix.eureka2.interests.ChangeNotifications.instanceInfoIdentityComparator;
+
+/**
+ * A special view of full registry, which provides both full {@link Applications} data
+ * structure, and delta changes that need to be applied to the previous copy of the
+ * {@link Applications} object to transition to the current value.
+ *
+ * @author Tomasz Bak
+ */
+public class Eureka2FullFetchWithDeltaView {
+
+    private static final Logger logger = LoggerFactory.getLogger(Eureka2FullFetchWithDeltaView.class);
+
+    private final Observable<ChangeNotification<InstanceInfo>> notifications;
+    private final long refreshIntervalMs;
+    private final Scheduler scheduler;
+
+    private final AtomicReference<RegistryFetch> latestCopy = new AtomicReference<>();
+    private final BehaviorSubject<RegistryFetch> latestCopySubject = BehaviorSubject.create();
+    private Subscription subscription;
+
+
+    public Eureka2FullFetchWithDeltaView(Observable<ChangeNotification<InstanceInfo>> notifications,
+                                         long refreshIntervalMs,
+                                         Scheduler scheduler) {
+
+        this.notifications = notifications;
+        this.refreshIntervalMs = refreshIntervalMs;
+        this.scheduler = scheduler;
+        connect();
+    }
+
+    /**
+     * Connect happens immediately after child object is constructed.
+     */
+    private void connect() {
+        subscription = notifications
+                .compose(ChangeNotifications.<InstanceInfo>delineatedBuffers())
+                .compose(emitAndAggregateChanges(
+                        instanceInfoIdentityComparator(),
+                        refreshIntervalMs, TimeUnit.MILLISECONDS,
+                        scheduler
+                ))
+                .map(new Func1<List<ChangeNotification<InstanceInfo>>, RegistryFetch>() {
+                    @Override
+                    public RegistryFetch call(List<ChangeNotification<InstanceInfo>> instanceInfos) {
+                        return updateSnapshot(instanceInfos);
+                    }
+                })
+                .subscribe(latestCopySubject);
```

Comment:
This will unsubscribe if there is no subscriber, and we want to cache it all the time, as building this view is very expensive.